arangod/RestServer/FlushFeature.cpp

```diff
@@ -57,42 +57,39 @@ namespace arangodb {
 class FlushFeature::FlushSubscriptionBase
     : public FlushFeature::FlushSubscription {
  public:
-  virtual Result commit(VPackSlice data) override final {
+  virtual Result commit(VPackSlice data, TRI_voc_tick_t tick) override final {
     if (data.isNone()) {
       // upgrade tick without commiting actual marker
-      resetCurrentTick(_engine.currentTick());
+      resetCurrentTick(tick);
       return {};
     }
 
-    return commitImpl(data);
+    return commitImpl(data, tick);
   }
 
   /// @brief earliest tick that can be released
-  virtual TRI_voc_tick_t tick() const = 0;
+  TRI_voc_tick_t tick() const noexcept {
+    return _tickPrevious;
+  }
 
  protected:
   FlushSubscriptionBase(
       std::string const& type, // subscription type
-      TRI_voc_tick_t databaseId, // vocbase id
-      arangodb::StorageEngine const& engine // vocbase engine
+      TRI_voc_tick_t databaseId // vocbase id
   ): _databaseId(databaseId),
-     _engine(engine),
      _tickCurrent(0), // default (smallest) tick for StorageEngine
      _tickPrevious(0), // default (smallest) tick for StorageEngine
      _type(type) {
-    // it's too early to use _engine.currentTick() here,
-    // storage engine may not be initialized yet
   }
 
   void resetCurrentTick(TRI_voc_tick_t tick) noexcept {
     _tickPrevious = _tickCurrent;
```

Comment:
Just wondering if this needs to be an atomic operation, or at least somehow protected against concurrent reads and writes of `_tickPrevious` and `_tickCurrent`. At least the `tick()` method is public and returns `_tickPrevious`. Is it possible that one thread is in `resetCurrentTick?` while another thread is in `tick()`?

Reply:
good point